src/features/Layout/Layout.tsx

```diff
@@ -139,6 +139,7 @@ export default function Layout({ config = {}, className = '', children }: Props)
   const { pathname } = useLocation();
 
   const [topHeight, setTopHeight] = useState(0);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
```

Comment:
The implementation is convoluted, Layout and other nav elements don't need to be aware of the Tray behaviour. Can we try using something like https://github.com/streamich/react-use/blob/master/docs/useClickAway.md (to import in `/vendor/react-use`)?